### PR TITLE
Update the `.distignore` file to exclude the ./client/admin folder from zip

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

p1729060714505489-slack-C01DT6U03HC

This PR fixes the issue where the zip file size increased significantly due to the inclusion of the `./client/admin` folder. The `.distignore` file is updated to exclude the `./client/admin` folder from the zip.

See: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/bin/build-zip.sh#L28-L31

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. cd `./plugins/woocommerce`
3. Run `pnpm build:zip`
4. Check the `woocomerce.zip` file size ~17MB
5. Unzip the file and check if the `./client/admin` folder is not included. Note: <= WC 9.4 contains the `./client/admin/config,` but it's actually not needed in the production zip file, so it's also excluded.
6. Confirm the Admin pages loaded properly

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Update the `.distignore` file to exclude the ./client/admin folder from zip

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/.distignore [hunk 1]
```diff
@@ -36,3 +36,4 @@ renovate.json
 tsconfig.*
 webpack.config.js
 client/legacy/
+client/admin/
```
Comment (rjchow) [inline plugins/woocommerce/.distignore]: leaving a note for later, to investigate if we need to negate the exclusion for feature flag

**Comment by chihsuan** [comment]
Hmm, just tested a bit more, and it seems that now `woocommerce/assets/client/admin` is missing in zip. 😓  I'll update the PR to fix it.

**Comment by chihsuan** [comment]
Fixed in a509a4a227e2b7a6680ef6f351ea062dd2873232.

**Review by rjchow — APPROVED** [review]
Hey @chihsuan I did indeed find that the feature flag config was being excluded. I couldn't seem to get rsync to work correctly with both --exclude and --include so I added a cp step below

**Comment by chihsuan** [comment]
> Hey @chihsuan I did indeed find that the feature flag config was being excluded. I couldn't seem to get rsync to work correctly with both --exclude and --include so I added a cp step below

Thanks for investigating! 👍 @rjchow  This is interesting. Why do we need these config files? I thought we only used them for development. 🤔

**Comment by rjchow** [comment]
@chihsuan that's a good question, I just assumed it's used because of `core.json` 👀

**Comment by rjchow** [comment]
I think you're right, the only reference I could find to it was in https://github.com/woocommerce/woocommerce/blob/f99b78d8520467f12b1db8cdb28621e898a01640/plugins/woocommerce/bin/generate-feature-config.php#L1 and that's only used in the build step and not at runtime 🤦 

I'll revert

**Comment by chihsuan** [comment]
@rjchow Thanks for confirming! That's what I was thinking, too. 🙌   I believe it's safe to remove configs from the zip as woocommerce-admin's [`.distignore`](https://github.com/woocommerce/woocommerce-admin/blob/main/.distignore#L30) file also excludes the `configs` directory. I think they were unintentionally included when the admin was merged into the core.

**Comment by rjchow** [comment]
I rolled back to check if they were being included previously and they were 🤣

**Pull request merged**